x-pack/plugins/canvas/public/components/embeddable_flyout/flyout.tsx

```diff
@@ -68,44 +84,17 @@ const mergeProps = (
       }
 
       addEmbeddable(pageId, partialElement);
-      ownProps.onClose();
+      restProps.onClose();
     },
-  };
+    [addEmbeddable, pageId, restProps]
+  );
+
+  return (
+    <EmbeddableFlyoutPortal
+      {...restProps}
+      {...state}
```

Comment:
This is dropping the entire state from redux in as props, which I don't think is necessary.

Reply:
Yes, I was not reading the doc carefully, It is doesn't need to pass all the `redux` state. Thanks for it.